compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/impl/FirPartiallyResolvedArgumentList.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.expressions.impl
+
+import org.jetbrains.kotlin.fir.FirSourceElement
+import org.jetbrains.kotlin.fir.declarations.FirValueParameter
+import org.jetbrains.kotlin.fir.expressions.FirArgumentList
+import org.jetbrains.kotlin.fir.expressions.FirExpression
+import org.jetbrains.kotlin.fir.visitors.FirTransformer
+import org.jetbrains.kotlin.fir.visitors.FirVisitor
+import org.jetbrains.kotlin.fir.visitors.transformInplace
+
+class FirPartiallyResolvedArgumentList internal constructor(
+    override var source: FirSourceElement?,
+    override val arguments: MutableList<FirExpression>,
+    private val matchedValueParameters: List<FirValueParameter?>
+) : FirArgumentList() {
```

Comment:
Any reason not to extend `FirAbstractArgumentList`? Then, you may avoid some overrides here.

Reply:
I am actually overriding all the members in `FirAbstractArgumentList` and provide a sensible implementation so this partial arg list can still function as a normal `FirArgumentList` for further transformations. I am not sure if it's necessary but it doesn't seem to hurt.